typescript/scripting-api/index.d.ts

```diff
@@ -31,47 +50,20 @@ declare namespace WoT {
      * Dictionary that represents the constraints for discovering Things as key-value pairs. 
      */
     export interface ThingFilter {
-        /**
-         * The method field represents the discovery type that should be used in the discovery process. The possible values are defined by the DiscoveryMethod enumeration that can be extended by string values defined by solutions (with no guarantee of interoperability). 
-         */
-        method?: DiscoveryMethod | string; // default value "any", DOMString
-        /**
-         * The url field represents additional information for the discovery method, such as the URL of the target entity serving the discovery request, such as a Thing Directory or a Thing.
-         */
-        url?: string;
-        /**
-         * The query field represents a query string accepted by the implementation, for instance a SPARQL query. 
-         */
-        query?: string;
         /**
          * The fragment field represents a template object used for matching against discovered Things.
          */
         fragment?: object;
     }
 
-    /** The DiscoveryMethod enumeration represents the discovery type to be used */
-    export enum DiscoveryMethod {
-        /** does not restrict */
-        "any",
-        /** for discovering Things defined in the same Servient */
-        "local",
-        /** for discovery based on a service provided by a Thing Directory */
-        "directory",
-        /** for discovering Things in the same/reachable network by using a supported multicast protocol */
-        "multicast"
-    }
-
     /**
      * The ThingDiscovery object is constructed given a filter and provides the properties and methods
      * controlling the discovery process. 
      */
-    export interface ThingDiscovery {
+    export interface ThingDiscoveryProcess extends AsyncIterable<ThingDescription> {
         filter?: ThingFilter;
```

Comment:
I have one more question about `filter` since it should be part of the constructor.

![image](https://github.com/w3c/wot-scripting-api/assets/11281461/69aff473-b673-42d0-877a-c95336bb786f)


For `ConsumedThing` we have a **_constructor_** as well and we do not have it as a variable there...
What do you think should we do here?

Reply:
Hmmm, good question, I was wondering the same, but I think in TypeScript interfaces, it is not possible to define a constructor, is it?

Reply:
My impression was that TypeScript and WebIDL interfaces simlpy differ in this regard, but I am really not sure here.

Reply:
Mhh, I just think we need to align both cases...

Having said that, since we cannot model constructors in TypeScript I think we should not mention/list `filter` at all here. 
Anyhow, I would like to hear comments from others as well. @relu91 @zolkis others?

Reply:
@danielpeintner I don't get the problem, as there are constructors in TypeScript. 
But the spec should be clear about mapping WebIDL object, ECMAScript class and TypeScript [class](https://www.typescriptlang.org/docs/handbook/2/classes.html).

Reply:
Classes have constructors, of course. What @JKRhb and I have been looking for are constructors on interfaces. Apparently even those exist but I have not seen them before...

https://www.typescriptlang.org/docs/handbook/interfaces.html#difference-between-the-static-and-instance-sides-of-classes

Reply:
> Apparently even those exist but I have not seen them before...

Me neither, actually, but using `new` seems to work :) I wonder if we should adjust the other TypeScript interfaces like this as well then?

Reply:
I think so.. we are using "constructor" for 2 interfaces. `ThingDiscoveryProcess` and `ConsumedThing`

Reply:
In order to make things a bit easier to track, I opened a separate PR for the `ConsumedThing` interface: https://github.com/w3c/wot-scripting-api/pull/515